controllers/discovery/initDiscoveryUserData.js

```diff
@@ -21,6 +21,26 @@ const InitDiscoveryUserData = async (req, res) => {
     FROM users A
     WHERE A.user_id != :userId AND A.is_anonymous = false AND A.is_banned = false`;
 
+    let user_topics = await sequelize.query(
+      `select tp.topic_id from topics as tp
+      left join user_topics as utp on tp.topic_id = utp.topic_id
+      where utp.user_id = :userId`,
+      {
+        type: QueryTypes.SELECT,
+        replacements: {
+          userId
+        }
+      }
+    );
+    let topicIds = user_topics.map((topic) => topic.topic_id);
+    const similarTopicQuery =
+      topicIds.length > 0
+        ? `ARRAY( select name from topics as tp
+          left join user_topics as utp on tp.topic_id = utp.topic_id
+          where utp.user_id = A.user_id and tp.topic_id in (:topicIds) limit 3
+        )`
+        : 'ARRAY[]::text[]';
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is fetching the user's topics and then constructing a query to fetch similar topics for other users. However, there is no error handling in case the database query fails. It would be good to wrap this in a try-catch block to handle any potential errors that might occur during the execution of the query.

```diff
+    try {
    let user_topics = await sequelize.query(
      `select tp.topic_id from topics as tp
      left join user_topics as utp on tp.topic_id = utp.topic_id
      where utp.user_id = :userId`,
      {
        type: QueryTypes.SELECT,
        replacements: {
          userId
        }
      }
    );
    let topicIds = user_topics.map((topic) => topic.topic_id);
    const similarTopicQuery =
      topicIds.length > 0
        ? `ARRAY( select name from topics as tp
          left join user_topics as utp on tp.topic_id = utp.topic_id
          where utp.user_id = A.user_id and tp.topic_id in (:topicIds) limit 3
        )`
        : 'ARRAY[]::text[]';
+    } catch (error) {
+        console.error(`Failed to execute query: ${error}`);
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->